djvm/src/main/kotlin/sandbox/Task.kt

```diff
@@ -21,7 +22,12 @@ class Task(private val function: SandboxFunction<in Any?, out Any?>?) : SandboxF
      * Strings and Enums, as well as for arrays of these types.
      */
     override fun apply(input: Any?): Any? {
-        return function?.apply(input?.sandbox())?.unsandbox()
+        val value = try {
+            function?.apply(input?.sandbox())
+        } catch (t: Throwable) {
+            throw t.escapeSandbox()
```

Comment:
Is this `catch` still related to the multi-type catch handling? Or is it addressing a separate issue?

Reply:
That `catch` is ensuring that an exception is "marshalled" correctly to a non-sandboxed type as it leaves so that we don't leak any references to `sandbox.*` classes. That marshalling includes the exception's `cause` and the contents of `getSuppressed()` too. (Suppressed exceptions being another JDK7 feature that this PR addresses.)